src/lints/enum_struct_variant_changed_kind.ron

```diff
@@ -0,0 +1,77 @@
+SemverQuery(
+    id: "enum_struct_variant_changed_kind",
+    human_readable_name: "An enum struct variant changed kind",
+    description: "A pub enum struct variant with at least one pub field changed kind",
+    required_update: Major,
+    lint_level: Deny,
+    // TODO: If the Rust reference gains a more detailed explanation of enum *variants*,
+    //       switch the explanation to point there instead. The current link isn't great.
+    reference_link: Some("https://doc.rust-lang.org/reference/items/enumerations.html"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Enum {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        enum_name: name @output @tag
+
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        variant {
+                            ... on StructVariant {
+                                kind: __typename @output @tag
+                                public_api_eligible @filter(op: "=", value: ["$true"])
+                                attrs @filter(op: "not_contains", value: ["$non_exhaustive"])
```

Comment:
Reading this more closely, I don't think we need this clause. Accessing the named field is breaking already, exhaustive or not.
```suggestion
```

Can you ensure the tests cover this case appropriately?